src/Controller/PageAdminController.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the ekino/sonata project.
+ *
+ * (c) Ekino
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\HelpersBundle\Controller;
+
+use Sonata\BlockBundle\Block\BlockServiceManager;
+use Sonata\BlockBundle\Model\Block;
+use Sonata\HelpersBundle\Block\BlockFilter\BlockFilter;
+use Sonata\PageBundle\Admin\BlockAdmin;
+use Sonata\PageBundle\Controller\PageAdminController as BasePageAdminController;
+use Sonata\PageBundle\Model\Page;
+use Sonata\PageBundle\Page\TemplateManagerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Exception\AccessDeniedException;
+
+/**
+ * Class PageAdminController.
+ *
+ * @author Vincent Laurier <vincent.laurier@ekino.com>
+ * @author Romain Mouillard <romain.mouillard@ekino.com>
+ * @author Ramesses Bonhof Keny <ramesses-bonhof.keny@ekino.com>
+ * @author Christian Kollross <christian.kollross@ekino.com>
+ */
+class PageAdminController extends BasePageAdminController
+{
+    /**
+     * @var BlockFilter
+     */
+    private $blockFilter;
+
+    /**
+     * PageAdminController constructor.
+     */
+    public function __construct(BlockFilter $blockFilter)
+    {
+        $this->blockFilter = $blockFilter;
+    }
+
+    /**
+     * @param Request|null $request
+     *
+     * @throws AccessDeniedException
+     * @throws NotFoundHttpException
+     *
+     * @return Response
+     */
+    public function composeContainerShowAction(Request $request = null)
+    {
+        /** @var BlockAdmin $blockAdmin */
+        $blockAdmin = $this->get('sonata.page.admin.block');
```

Comment:
we have to find a way to inject it. no more direct get from the container

Reply:
Well, `<argument type="service" id="sonata.page.admin.block" />` doesn't work.

Reply:
Sorry, my bad. It works if you assign the BlockAdmin in the constructor... :innocent: